### PR TITLE
Remove the IAM Policy Controller

### PR DESCRIPTION
**Pull request by yiraeChristineKim**
- Ensure that the governance-policy-addon-controller removes the existing ManifestWork deploying the IAM policy controller. This should lead to the IAM policy controller being removed.

- Delete all iam-policy-controller deploying procedure

Ref: https://issues.redhat.com/browse/ACM-10859

### CODE REVIEW

File: config/rbac/role.yaml [hunk 1]
```diff
@@ -17,6 +17,7 @@ rules:
   resources:
   - clustermanagementaddons
   verbs:
+  - delete
```
[the remark this answers is not in the document]
Comment (dhaiducek) [reply config/rbac/role.yaml:20]: Actually, maybe add another item that just has `delete` for `iam-policy-controller`. That's probably the most targeted.

**Comment by dhaiducek** [comment]
> Should I add an e2e- test about cleaning clusteramangementAddon?

Yes, I think so.

**Comment by dhaiducek** [comment]
> /hold while we discuss this with the Hub installer team on if multicluster-hub should handle deleting the CMA.

@yiraeChristineKim Would you move the `delete` changes over to a separate PR? The other changes could get merged as-is in my opinion.

File: pkg/addon/deleteaddon/deleate_addon.go [hunk 2]
```diff
@@ -0,0 +1,36 @@
+package deleteaddon
```
Comment (mprahl) [inline pkg/addon/deleteaddon/deleate_addon.go:1]: Could you please delete this file now that the uninstall is happening in the multiclusterhub-operator?

File: main.go [hunk 3]
```diff
@@ -50,17 +49,17 @@ import (
 //+kubebuilder:rbac:groups=certificates.k8s.io,resources=certificatesigningrequests;certificatesigningrequests/approval,verbs=get;list;watch;create;update
 //+kubebuilder:rbac:groups=certificates.k8s.io,resources=signers,verbs=approve
 //+kubebuilder:rbac:groups=cluster.open-cluster-management.io,resources=managedclusters,verbs=get;list;watch
-//+kubebuilder:rbac:groups=addon.open-cluster-management.io,resources=clustermanagementaddons,verbs=get;list;watch
+//+kubebuilder:rbac:groups=addon.open-cluster-management.io,resources=clustermanagementaddons,verbs=get;list;watch;delete
```
Comment (mprahl) [inline main.go:52]:
```suggestion
//+kubebuilder:rbac:groups=addon.open-cluster-management.io,resources=clustermanagementaddons,verbs=get;list;watch
```

**Review by mprahl — CHANGES_REQUESTED** [review]
The permission change can be reverted back now.

File: main.go (hunk 3 above)
Comment (yiraeChristineKim) [reply main.go]: done

**Review by mprahl — APPROVED** [review]

**Review by dhaiducek — APPROVED** [review]

**Pull request merged**